libs/model-assessment/src/lib/ModelAssessmentDashboard/Controls/ModelOverview/StatsTableUtils.ts

```diff
@@ -0,0 +1,116 @@
+
+import { ErrorCohort, ILabeledStatistic } from "@responsible-ai/core-ui";
+import { PointOptionsObject } from "highcharts";
+import { IDropdownOption } from "office-ui-fabric-react";
+
+export function generateCohortsStatsTable(
+  cohorts: ErrorCohort[],
+  selectableMetrics: IDropdownOption[],
+  labeledStatistics: ILabeledStatistic[][],
+  selectedMetrics: string[]
+) {
+  const items: PointOptionsObject[] = cohorts.map(
+    (errorCohort, cohortIndex) => {
+      return {
+        colorValue: 0,
+        value: errorCohort.cohortStats.totalCohort,
+
+        x: 0,
+        // metric index for Count column
+        y: cohortIndex
+      };
+    }
+  );
+  selectableMetrics
+    .filter((element: IDropdownOption) =>
+      selectedMetrics.includes(element.key.toString())
+    )
+    .forEach((metricOption: IDropdownOption, metricIndex: number) => {
+      // first determine min and max values
+      let metricMin = Number.MAX_SAFE_INTEGER;
+      let metricMax = Number.MIN_SAFE_INTEGER;
+      cohorts.forEach((_errorCohort, cohortIndex) => {
+        const stat = labeledStatistics[cohortIndex].find(
+          (stat) => stat.key === metricOption.key
+        );
+        if (stat) {
+          if (stat.stat > metricMax) {
+            metricMax = stat.stat;
+          }
+          if (stat.stat < metricMin) {
+            metricMin = stat.stat;
+          }
+        }
+      });
+      // use min and max to normalize the colors
+      const metricMinMaxDiff = metricMax - metricMin;
+      cohorts.forEach((_errorCohort, cohortIndex) => {
+        const stat = labeledStatistics[cohortIndex].find(
+          (stat) => stat.key === metricOption.key
+        );
+        if (stat && !Number.isNaN(stat.stat)) {
+          items.push({
+            colorValue: (stat.stat - metricMin) / metricMinMaxDiff,
+            value: Number(stat.stat.toFixed(3)),
+            x: metricIndex + 1,
+            y: cohortIndex
+          });
+        } else {
+          // not a numeric value (NaN), so just put null
+          items.push({
+            color: "#808080",
+            value: Number.NaN,
+            x: metricIndex + 1,
+            y: cohortIndex // gray
+          });
+        }
+      });
+    });
+  return items;
+}
+
+export function wrapYAxisLabels(label: string, wrapOnWhitespace = false) {
+  const maxLineLength = 40;
+  const maxLines = 2;
+  const lineWrapHtmlTag = "<br />";
+  const ellipsis = "...";
+
+  // check if there are suitable spots for a linewrap
+  // if not just wrap after maxLineLength characters
+  let slicingIndex = maxLineLength;
+  const closingParenthesis = ") ";
+  const openingParenthesis = " (";
+  const whitespace = " ";
+  const searchString = new Set(
+    label.slice(slicingIndex - maxLineLength / 2, slicingIndex)
+  );
+
+  if (searchString.has(closingParenthesis)) {
+    // option 1: wrap after closing parenthesis
+    slicingIndex =
+      label.indexOf(closingParenthesis, slicingIndex - maxLineLength / 2) + 2;

```

Comment:
I think this code could be refactored a bit more, maybe like:

```
let axisLabelString = ""
if (searchString.has(closingParenthesis)) {
  axisLabelString = closingParenthesis
} else if (searchString.has(openingParenthesis)) {
  axisLabelString = openingParenthesis
} else if (wrapOnWhitespace && searchString.has(whitespace)) {
  axisLabelString = whitespace
}
if (axisLabelString != "") {
  slicingIndex = label.indexOf(axisLabelString, slicingIndex - maxLineLength / 2) + 1;
  if (axisLabelString == closingParenthesis) {
    slicingIndex += 1
  }
}
```
or maybe move the code:
```
def updateSlicingIndex(slicingIndex...): number => {
  label.indexOf(axisLabelString, slicingIndex - maxLineLength / 2) + 1;
  if (axisLabelString == closingParenthesis) {
    slicingIndex += 1
  }
}
```
into a function and call that in each of the if/else if's above

Reply:
You may have noticed that the index is not just `indexOf(...) + 1` in every case. It depends on the slicing position, too. I personally feel like it's clearer to determine the index first but I agree that there's some redundancy. We had a design meeting today in which we decided to simplify these cohort descriptions a bit, so I'll be able to remove the parenthesis cases altogether. That should make the code simple enough to avoid needing a separate function.

Reply:
Updated this now!